nx/lib/nx.ex

```diff
@@ -3265,6 +3265,129 @@ defmodule Nx do
     end)
   end
 
+  @doc ~S"""
+  Split a tensor into train and test subsets.
+
+  `split` split must be an integer greater than zero and less than the length of the tensor.
+
+  ## Options
+
+    * `:axis` - The axis along which to take the values from. Defaults to `0`.
+
+  ## Examples
+
+    Split a tensor into two separate tensors.
+
+    iex> {train, test} = Nx.split(Nx.tensor([[3, 6, 5], [26, 75, 3], [23, 4, 1]]), 2, axis: 0)
+    iex> train
+    #Nx.Tensor<
+      s64[2][3]
+      [
+        [3, 6, 5],
+        [26, 75, 3]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[1][3]
+      [
+        [23, 4, 1]
+      ]
+    >
+
+    iex> {train, test} = Nx.split(Nx.tensor([[3, 6, 5], [26, 75, 3], [23, 4, 1]]), 2, axis: 1)
+    iex> train
+    #Nx.Tensor<
+      s64[3][2]
+      [
+        [3, 6],
+        [26, 75],
+        [23, 4]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[3][1]
+      [
+        [5],
+        [3],
+        [1]
+      ]
+    >
+
+    iex> {train, test} = Nx.split(Nx.tensor([[3, 6, 5, 20], [26, 75, 3, 9], [23, 4, 1, 5]], names: [:rows, :columns]), 2, axis: :columns)
+    iex> train
+    #Nx.Tensor<
+      s64[rows: 3][columns: 2]
+      [
+        [3, 6],
+        [26, 75],
+        [23, 4]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[rows: 3][columns: 2]
+      [
+        [5, 20],
+        [3, 9],
+        [1, 5]
+      ]
+    >
+
+    iex>{train, test} = Nx.split(Nx.tensor([[3, 6, 5, 20], [26, 75, 3, 9], [23, 4, 1, 5]], names: [:rows, :columns]), 2, axis: :rows)
+    iex> train
+    #Nx.Tensor<
+      s64[rows: 2][columns: 4]
+      [
+        [3, 6, 5, 20],
+        [26, 75, 3, 9]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[rows: 1][columns: 4]
+      [
+        [23, 4, 1, 5]
+      ]
+    >
```

Comment:
I think these are good tests, but we don't need them as examples. You can move them to the test file

Reply:
(these == the named axis examples)